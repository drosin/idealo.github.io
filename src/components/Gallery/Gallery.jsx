import React from 'react';
import './Gallery.scss';

import GalleryService from '../../services/GalleryService';
import ProgressiveImage from '../../components/ProgressiveImage/ProgressiveImage';

export default (props) => {
    function renderImages() {
        let images = [];

        GalleryService.getImages().forEach((image, index) => {
            images.push(
                <ProgressiveImage keepRatio={true} className='gallery__image' key={'galleryImage' + index} src={image.imgUrl} thumbnail={image.thumbnail} blurRadius='3' />
            );
        });

        return images;
    }

    return (
        <div className='gallery__container'>
            {renderImages()}
        </div>
    );

}