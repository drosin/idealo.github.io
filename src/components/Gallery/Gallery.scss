.gallery {
    &__container {
        display: flex;
        flex-wrap: wrap;

        .progressiveImage__original {
            transition: opacity 1s, filter .3s;

            &:hover {
                filter: brightness(100%) !important;
            }
        }

        &:hover .progressiveImage__original {
            filter: brightness(80%);
        }
    }



    &__image {
        width: 25%;
        @media screen and (max-width: 750px) {
            width: 50%;
        }
    }
}