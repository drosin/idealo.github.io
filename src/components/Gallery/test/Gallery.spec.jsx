import React from 'react';
import renderer from 'react-test-renderer';
import Gallery from '../Gallery';

jest.mock('../../../components/ProgressiveImage/ProgressiveImage', () => 'ProgressiveImage')

describe('Gallery component', () => {
    it('should render as expected', () => {
        const renderedComponent = renderer.create(
            <Gallery />
        );
        expect(renderedComponent.toJSON()).toMatchSnapshot();
    })
});