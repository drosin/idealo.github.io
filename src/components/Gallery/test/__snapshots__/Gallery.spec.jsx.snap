// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery component should render as expected 1`] = `
<div
  className="gallery__container"
>
  <ProgressiveImage
    blurRadius="3"
    className="gallery__image"
    keepRatio={true}
    src="static/jobs_04.png"
    thumbnail="static/thumbnails/jobs_04.png"
  />
  <ProgressiveImage
    blurRadius="3"
    className="gallery__image"
    keepRatio={true}
    src="static/jobs_08.png"
    thumbnail="static/thumbnails/jobs_08.png"
  />
  <ProgressiveImage
    blurRadius="3"
    className="gallery__image"
    keepRatio={true}
    src="static/jobs_01.png"
    thumbnail="static/thumbnails/jobs_01.png"
  />
  <ProgressiveImage
    blurRadius="3"
    className="gallery__image"
    keepRatio={true}
    src="static/jobs_03.png"
    thumbnail="static/thumbnails/jobs_03.png"
  />
  <ProgressiveImage
    blurRadius="3"
    className="gallery__image"
    keepRatio={true}
    src="static/jobs_02.png"
    thumbnail="static/thumbnails/jobs_02.png"
  />
  <ProgressiveImage
    blurRadius="3"
    className="gallery__image"
    keepRatio={true}
    src="static/jobs_05.png"
    thumbnail="static/thumbnails/jobs_05.png"
  />
  <ProgressiveImage
    blurRadius="3"
    className="gallery__image"
    keepRatio={true}
    src="static/jobs_07.png"
    thumbnail="static/thumbnails/jobs_07.png"
  />
  <ProgressiveImage
    blurRadius="3"
    className="gallery__image"
    keepRatio={true}
    src="static/jobs_06.png"
    thumbnail="static/thumbnails/jobs_06.png"
  />
</div>
`;
